sources/lib/structures/typed.ml

```diff
@@ -79,6 +83,9 @@ and 'a tt_desc =
   | TTnamed of Hstring.t * ('a tterm, 'a) annoted
   | TTite of ('a tform, 'a) annoted *
              ('a tterm, 'a) annoted * ('a tterm, 'a) annoted
+  | TTsharp of bool * ('a tterm, 'a) annoted  * Hstring.t
```

Comment:
Could this be named something like projection ? The `sharp` reference is pretty specific to smt-lib and doesn't really help understand what this construction does.